misc/dbt-materialize/dbt/include/materialize/macros/columns.sql

```diff
@@ -22,3 +22,28 @@
         {{ select_sql }}
     ) as __dbt_sbq
 {% endmacro %}
+
+{% macro materialize__get_empty_schema_sql(columns) %}
+    {%- set col_err = [] -%}
+    {%- set col_naked_numeric = [] -%}
+    select
+    {% for i in columns %}
+      {%- set col = columns[i] -%}
+      {%- if col['data_type'] is not defined -%}
+        {%- do col_err.append(col['name']) -%}
+      {#-- If this column's type is just 'numeric' then it is missing precision/scale, raise a warning --#}
+      {%- elif col['data_type'].strip().lower() in ('numeric', 'decimal', 'number') -%}
+        {%- do col_naked_numeric.append(col['name']) -%}
+      {%- endif -%}
+      {% set col_name = adapter.quote(col['name']) if col.get('quote') else col['name'] %}
+      -- NOTE(morsapaes): in a future release of dbt, the default macro will use
+      -- the global cast macro, as modified here, and we can remove this custom
+      -- override. See: https://github.com/dbt-labs/dbt-adapters/pull/165
+      {{ cast("null", col['data_type']) }} as {{ col_name }}{{ ", " if not loop.last }}
+    {%- endfor -%}
+    {%- if (col_err | length) > 0 -%}
+      {{ exceptions.column_type_missing(column_names=col_err) }}
+    {%- elif (col_naked_numeric | length) > 0 -%}
+      {{ exceptions.warn("Detected columns with numeric type and unspecified precision/scale, this can lead to unintended rounding: " ~ col_naked_numeric ~ "`") }}
+    {%- endif -%}
+{% endmacro %}
```

Comment:
To confirm my understanding: this file will go away after the 1.8 upgrade merges? Are you planning to wait until the 1.8 upgrade merges before merging this PR?

Reply:
Yes, because it depends on the new package structure in dbt v1.8. The upgrade PR is ready to merge, with a couple of follow-up TODOs.

Reply:
And the version of the macro above was already released in `dbt-adapters`, yeah. This is now released independently to `dbt-core`, after the refactoring on their end.